protocols/gossipsub/src/config.rs

```diff
@@ -73,9 +73,13 @@ pub struct GossipsubConfig {
     /// When set to `true`, prevents automatic forwarding of all received messages. This setting
     /// allows a user to validate the messages before propagating them to their peers. If set to
     /// true, the user must manually call `propagate_message()` on the behaviour to forward message
-    /// once validated (default is false).
+    /// once validated (default is `false`).
     pub manual_propagation: bool,
 
+    /// Message signing is on by default.  When this parameter is set,
+    /// published messages are not signed by the libp2p key.
+    pub disable_message_signing: bool,
```

Comment:
Do I understand correctly that one has to pass a keypair to `GossipSub::new` even when setting `disable_message_signing` to `true`?

Wouldn't something along the following lines work as well?

```suggestion
    pub enum Signing {
        Enabled(Keypair),
        Disabled(PeerId),
    }
    
    // ...

    pub disable_message_signing: Signing,
```

Reply:
This is interesting. You are correct in that currently we unnecessarily require the keypair. 
I'm happy to go with whatever you think will be the best ergonomics to use this. 

If we create a `Signing` enum in the config, the `config()` will no longer implement `Default` and a user will be required to enter a `Signing` variant during the config initialisation. 

Alternatively, I could take signing out of the config, and take an Option<Keypair> when creating the main struct, however this may not be intuitive when initialising the struct that you are optionally invoking signing. 

Unless I'm mistaken you are suggesting something like:
```rust
let config = GossipsubConfig::new(Signing::Enabled(keypair));
Gossipsub::new(config);
```

Which seems fine. I'll add it in and we can change later if we need.

Reply:
After writing this, and attempting to integrate the new version, I realise it's really kind of an annoyance to not have `Default` implemented for the configuration, or to not allow the configuration be instantiated without a `PeerId` or `Keypair`.

The two solutions I see immediately are: 
1. Move the signing enum to the initialisation of `Gossipsub`. i.e `Gossipsub::new(Signing, GossipsubConfig)`
2. Add a default Signing variant, which could be Signing::Disabled(IDENTITY_PEERID). 

Both of these don't seem great.

Reply:
I went with 1.